### ISSUE TITLE
Create a static wrapper type (or value object) similar to Guid for strong typing

### ISSUE THREAD

**Issue by BlackBearFTW** [sqids/sqids-dotnet#37]
It would be great if this implementation could be wrapped in a static class similar to Guid so strong typing is allowed inside the domain.

**Comment by aradalvand** [comment]
This can already be done in userland by declaring your own custom types, or better yet, using things like [Vogen](https://github.com/SteveDunn/Vogen) (which is a great library).

```cs
[ValueObject<int>]
public readonly partial struct SqidsId;
```

Could you elaborate on what need you specifically have and how it's not already doable by defining your own types?

**Comment by BlackBearFTW** [comment]
> This can already be done in userland by declaring your own custom types, or better yet, using things like [Vogen](https://github.com/SteveDunn/Vogen) (which is a great library).
> 
> ```cs
> [ValueObject<int>]
> public readonly partial struct SqidsId; // you could also add methods like `Validate`, static factories, etc.
> ```
> 
> Could you elaborate on what need you specifically have and how it's not already doable by defining your own types?

I am familiar with Vogen and was originally trying to use that, but that would require a whole lot of additional code to handle model binding, json conversion and other stuff since your primitive will be an integer, while publicly facing (and in ToString) it should be displayed as string.

**Comment by aradalvand** [comment]
I have to ask again: "How is this not already doable by defining your own types"?

**Comment by BlackBearFTW** [comment]
Yes of course I can implement it myself, but that is with anything, I could write my own implementation of sqid, I could write my own implementation of value objects etc, the ability for me being able to do it myself isn't the point. 

I think this feature would improve the developer experience because they don't have to do it themselves. Besides strongly typed is something that should be strived for anyway, it makes it easier to recognise if a property is just a string or if its supposed to represent a sqid.

**Comment by kpietraszko** [comment]
I think it's a good point. For example, a library that's a bit similar: https://github.com/Cysharp/Ulid  
provides a type representing the id, which stores the underlying data (raw bytes in their case, but in sqids it would be numbers), along with methods to encode to string.  
This also allows them to provide integrations with System.Text.Json etc, see https://github.com/Cysharp/Ulid#integrate
